### PR TITLE
Plotter callback

### PR DESCRIPTION
**Pull request by aleju**
This pull request adds a class `PlotGenerator` in `keras/utils/plotting_utils.py` (new file) and a callback `Plotter` to `keras/callbacks.py`. The `Plotter` callback inherits from the `History` callback and just sends history data for loss and accuracy to the `PlotGenerator`. The latter one then generates a plot that shows the training loss, validation loss, training accuracy and validation accuracy over time/epochs. The plot is automatically updated at the end of each epoch. The plot can be shown in a window and/or saved to a file. Regression is used by to predict the next couple of values for each line.

I added the plotter callback to both mnist examples.

Example image (mnist):
plot2
I deactivated regression in that one, because it's not very helpful when values get close to 100% accuracy.

Example usage:

```python
from keras.callbacks import Plotter
# standard plot
model.fit(X_train, Y_train, callbacks=[Plotter()])
```

```python
from keras.callbacks import Plotter
# Dont show the plot window, but save to a file
model.fit(X_train, Y_train, callbacks=[Plotter(show_plot_window=False, save_to_filepath="/tmp/last_plot.png")])
```

```python
from keras.callbacks import Plotter
# No regression
model.fit(X_train, Y_train, callbacks=[Plotter(show_regressions=False)])
```

### CODE REVIEW

**Comment by bayesrule** [comment]
Looks cool! But is there an option to disable accuracy plotting? I'm dealing with regression tasks most of the time instead of classification, in that scenario reported accuracy would be 0.

**Comment by aleju** [comment]
Currently in that case the accuracy chart on the right simply stays empty. There's no flag yet to hide it completely.

**Comment by fchollet** [comment]
The main issue I have with this type of plotting is that it would only be viable when you run your experiments on a machine that has a graphical desktop environment. It's one of the reasons why I would prefer visualization tools to work over the web. Run your experiment on any machine / cluster that has access to the internet, monitor it from any machine that has access to the internet. 

Another issue is that the API is both very complex and insufficiently complete. 

```python
keras.callbacks.Plotter(save_to_filepath=None, show_plot_window=True, linestyles=None, 
linestyles_first_epoch=None, show_regressions=True, poly_forward_perc=0.1, 
poly_backward_perc=0.2, poly_n_forward_min=5, poly_n_backward_min=10, poly_degree=1)
```

That's a lot of arguments (10), and only 3 of them are readily interpretable to me. That's a big UX problem. 

Yet, important workflow considerations like showing or hiding accuracy plots are not included.

Thoughts?

**Comment by jfsantos** [comment]
Using [Bokeh](https://github.com/bokeh/bokeh) could be a better option for a plotting monitor that could be checked over the web. 

The large amount of parameters for regression in the API is indeed not ideal and suggests maybe that it should be decoupled from the plotting object somehow.

**Comment by bayesrule** [comment]
+1 for Bokeh

**Comment by fchollet** [comment]
Developing visualization tools on top of Bokeh is a possibility (it was the first name mentioned when we started discussing visualization a couple month ago); if we are going to do that, it would be in a separate visualization app that plugs into Keras. You can join the discussion here: https://github.com/fchollet/keras/issues/254

**Comment by fchollet** [comment]
Closing this as we won't merge it. I do believe this is very valuable code for anybody who does they computation on a desktop with a GDE (e.g. most people who use their own GPU), but I don't think it belongs in master.
